File: Greedy, BruteForce/Java/10610 30.java
import java.util.*;
import java.util.Arrays;

public class Main {
    
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        String num = sc.nextLine();
        String[] arr = num.split("");
        Arrays.sort(arr, Collections.reverseOrder());
        String snum = String.join("", arr);
        
        int sum = 0;
        if(snum.charAt(snum.length()-1) != '0') System.out.println(-1);
        else {
            for(int i = 0 ; i < snum.length()-1; i++) {
                sum += snum.charAt(i) - '0';
            }
            
            if(sum % 3 == 0) System.out.println(snum);
            else System.out.println(-1);
        }
    }
}

File: mvc1/_etc/Kotlin/10610 30.kt
import java.util.*;

fun main(args: Array<String>) {
    
    var sc = Scanner(System.`in`)
    var s = sc.nextLine().split("").sorted().reversed().joinToString("")
    var sum = 0;
    if(s[s.length - 1] != '0') print(-1)
    else {
        for(i in 0 until s.length-1) {
            sum += s[i] - '0';
        }
        if(sum % 3 == 0) print(s);
        else print(-1);
    }

}
